### PR TITLE
Refactor:Component Layout Less to cssinjs

### PR DESCRIPTION
**Pull request by zpc7**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Layout support cssinjs      |
| 🇨🇳 Chinese |     Layout 支持 cssinjs      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/layout/style/index.tsx [hunk 1]
```diff
@@ -1,2 +1,239 @@
...
+// deps-lint-skip-all
+// import '../../style/index.less';
+// import './index.less';
+import { TinyColor } from '@ctrl/tinycolor';
+import type { CSSObject } from '@ant-design/cssinjs';
+import { genComponentStyleHook, mergeToken } from '../../_util/theme';
+import type { GenerateStyle, FullToken } from '../../_util/theme';
+import genLayoutLightStyle from './light';
+
+export interface LayoutToken extends FullToken<'Steps'> {
```
Comment (zombieJ) [inline components/layout/style/index.tsx:10]: `FullToken<'Steps'>` => `FullToken<'Layout'>`

File: components/layout/style/index.tsx [hunk 2]
```diff
@@ -1,2 +1,239 @@
...
+    layoutHeaderBackground,
+    layoutHeaderHeight: 64, // FIXME: hardcode in v4
+    layoutHeaderPadding: '0 50px', // FIXME: hardcode in v4
+    layoutHeaderColor: colorText,
+    layoutFooterPadding: '24px 50px', // FIXME: hardcode in v4
+    layoutFooterBackground: layoutBodyBackground,
+    layoutSiderBackground: layoutHeaderBackground,
+    layoutTriggerHeight: 48, // FIXME: hardcode in v4
+    layoutTriggerBackground: '#002140', // FIXME: hardcode in v4
+    layoutTriggerColor: '#Fff', // FIXME: hardcode in v4
```
Comment (zombieJ) [inline components/layout/style/index.tsx:229]: `#fff` 大小写，下面两个也是~

File: components/layout/style/index.tsx (hunk 1 above)
Comment (zpc7) [reply components/layout/style/index.tsx]: updated

File: components/layout/style/index.tsx (hunk 2 above)
Comment (zpc7) [reply components/layout/style/index.tsx]: updated

**Pull request merged**